host-provision.yml

```diff
@@ -0,0 +1,71 @@
+# Copyright 2020 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+---
+
+# Assumption: 
+# First time connection is not through public private key pair
+# FIrst time connection is always through customeradmin credentials password authentication
+# Consequentially, to start with, inventory file will not have details about SSH private key
+
+# Logic: 
+# First using the customeradmin credentials, keypair is generated
+# The tasks in host provisioning will be done using customeradmin
+# For running subsequent install-oracle.sh, customer can then ssh using the ansible user created
+# Inventory file may be updated with: ansible_ssh_private_key_file and ansible_ssh_user=ansible
+# Prior to calling the next steps (i.e.: install-oracle.sh)
+
+- name: Create private public key pair locally
+  hosts: localhost
+  tasks:
+  - include_role:
+      name: host-provision
+      tasks_from: ssh-keygen.yml
+  tags: host-provision
+
+- name: Create user and transfer public key to set up ssh equivalence
+  hosts: all
+  vars_prompt:
+    - name: ansible_password
+      prompt: Enter customeradmin password
```

Comment:
Nit: s/customeradmin/{{ firsttime_connect_user }}/

Reply:
done

Reply:
The latest round of changes have been incorporated and tested successfully; the following cmd line options is how it is now invoked:
```bash
✔ ~/DriveFS/My Drive/bmaas/host_provisioning/bms-toolkit [host-provision|✚ 2] 
15:48 $ runlocalssh ./host-provision.sh --instance-ip-addr 172.16.30.1  --instance-ssh-user ansible9
```

Please check from your end.
Thanks